### PR TITLE
Indent code snippets using `js-beautify`

### LINKED ISSUE #5245: Fix inconsistent whitespace & indentation in macro-generated examples markup
Recent work on macros in PRs such as #5213 , #5242 , and #5212 revealed some problems rendering markup generated by macros within example code snippets. They often have a large amount of excess whitespace and incorrect tabbing.

image

Using Jinja [whitespace control](https://jinja.palletsprojects.com/en/3.0.x/templates/#whitespace-control) helps a lot with eliminating excess newlines, but does not help with indentation.

We should find a way to make the indentation of our HTML (and possibly JS) for examples consistent. As a stretch goal, it would be nice to not have to worry as much about handling newlines, and have that handled for us as well.

### PR DESCRIPTION
**Pull request by jmuzina**
## Done

Formats all HTML, JS, and CSS code shown in `pre` code snippet blocks using [js-beautify](https://www.npmjs.com/package/js-beautify)

Fixes #5245, [WD-13629](https://warthogs.atlassian.net/browse/WD-13629)

## QA

- Look at some example code blocks and make sure they are indented with 2-space indentation. 
- [Full width dropdown navigation](https://vanilla-framework-5259.demos.haus/docs/patterns/navigation#full-width-dropdowns) is a good example to check as it has HTML, CSS, and JS, and was made with macros that generate some incorrect indentation in [main](https://staging.vanillaframework.io/docs/patterns/navigation#full-width-dropdowns)

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix release (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

### Before
image
### After
image

[WD-13629]: https://warthogs.atlassian.net/browse/WD-13629?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: templates/_layouts/docs.html [hunk 1]
```diff
@@ -126,6 +126,9 @@
       window.Prism = window.Prism || {};
       Prism.manual = true;
     </script>
+    <script defer src="https://cdnjs.cloudflare.com/ajax/libs/js-beautify/1.15.1/beautify.min.js"></script>
```
[the remark this answers is not in the document]
Comment (jmuzina) [reply templates/_layouts/docs.html:129]: CodeQL doesn't like using these CDN links - maybe we wan use rawgit or would have to use NPM instead @pastelcyborg

[the remark this answers is not in the document]
Comment (jmuzina) [reply templates/_layouts/docs.html:129]: js-beautify doesn't provide a SHA for an integrity check - but we could make it ourselves i think

**Comment by jmuzina** [comment]
Here's my initial implementation @bartaz @pastelcyborg  - any thoughts? 
Feel free to edit the branch as you see fit @pastelcyborg as you're also interested in this problem

File: templates/_layouts/docs.html [hunk 2]
```diff
@@ -126,6 +126,9 @@ <h4 class="p-table-of-contents__header">On this page:</h4>
       window.Prism = window.Prism || {};
       Prism.manual = true;
     </script>
+    <script defer src="https://cdnjs.cloudflare.com/ajax/libs/js-beautify/1.15.1/beautify.min.js" integrity="sha512-kElKX79OYM/vf2AU+lF4Zsi3AnKds70Mc6fRBImpwQzdZNbju6SyeTBBOUe1N9q1P456v1F01HAsjjWZkXIS4Q==" crossorigin="anonymous"></script>
```
Comment (bartaz) [inline templates/_layouts/docs.html:129]: Would be good to copy those to our static folder, not to rely on external resources.

Comment (jmuzina) [reply templates/_layouts/docs.html]: Updated to copy to static folder

Comment (bartaz) [reply templates/_layouts/docs.html]: No, it's fine as long as they are somewhere in our static files. Good to keep it consistent with the rest of files we are copying over from node modules.

**Comment by jmuzina** [comment]
> The only change I might make would be to either document that the functions come from a global/window scope or actually reference `window`, since otherwise the script calling the library functions might fail compilation/bundling in some scenarios.

Good call, updated this to reference `window` when calling the beautify functions

**Pull request merged**